packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js

```diff
@@ -99,6 +101,25 @@ export default class TelecomPackMigrationBuildingDetailsCtrl {
     }
   }
 
+  /* Check OTP reference value is valid */
+  isOtpRegexValid() {
+    this.otpReferenceValid = this.model.ptoReference.match(
+      /^[A-Z]{2}-[A-Z\d]{4}-[A-Z\d]{4}$/,
+    );
+  }
+
+  /**
+   * Check if OTP reference is valid
+   * @returns boolean
+   */
+  isOtpValid() {
+    return (
+      this.model.ptoReference != null &&
+      this.model.ptoReference !== '' &&
```

Comment:
I think your regex manage these case null and ''

Reply:
no it's not the same need.